openedx_tagging/core/tagging/rest_api/v1/views.py

```diff
@@ -144,3 +158,77 @@ def perform_create(self, serializer):
         Create a new taxonomy.
         """
         serializer.instance = create_taxonomy(**serializer.validated_data)
+
+
+class ObjectTagView(ReadOnlyModelViewSet):
+    """
+    View to retrieve paginated ObjectTags for a provided Object ID (object_id).
+
+    **Retrieve Parameters**
+        * object_id (required): - The Object ID to retrieve ObjectTags for.
+
+    **Retrieve Query Parameters**
+        * taxonomy (optional) - PK of taxonomy to filter ObjectTags for.
+        * page (optional) - Page number of paginated results.
+        * page_size (optional) - Number of results included in each page.
+
+    **Retrieve Example Requests**
+        GET api/tagging/v1/object_tags/:object_id
+        GET api/tagging/v1/object_tags/:object_id?taxonomy=1
+        GET api/tagging/v1/object_tags/:object_id?taxonomy=1&page=2
+        GET api/tagging/v1/object_tags/:object_id?taxonomy=1&page=2&page_size=10
+
+    **Retrieve Query Returns**
+        * 200 - Success
+        * 400 - Invalid query parameter
+        * 403 - Permission denied
+
+    **Create Query Returns**
+        * 403 - Permission denied
+        * 405 - Method not allowed
+
+    **Update Query Returns**
+        * 403 - Permission denied
+        * 405 - Method not allowed
+
+    **Delete Query Returns**
+        * 403 - Permission denied
+        * 405 - Method not allowed
+    """
+
+    serializer_class = ObjectTagSerializer
+    permission_classes = [ObjectTagObjectPermissions]
+    pagination_class = ObjectTagPagination
+    lookup_field = "object_id"
+
+    def get_queryset(self):
+        """
+        Return a queryset of object tags for a given object.
+
+        If a taxonomy is passed in, object tags are limited to that taxonomy.
+        """
+        object_id = self.kwargs.get("object_id")
+        query_params = ObjectTagListQueryParamsSerializer(
+            data=self.request.query_params.dict()
+        )
+        query_params.is_valid(raise_exception=True)
+        taxonomy = query_params.data.get("taxonomy", None)
+        taxonomy = get_taxonomy(taxonomy)
+        return get_object_tags(object_id, taxonomy)
```

Comment:
Hmm.. we don't really need the `taxonomy` data here, we're just fetching it to pass it to the API.
Could `get_object_tags` accept a `taxonomy_id` instead of an object, so we can just pass the ID through?

Reply:
Makes sense, I changed `get_object_tags` to accept `taxonomy_id`, had to update a few tests related to the API, but all seems to be working well.